lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -1007,10 +1007,7 @@ struct LoadOpConversion
 
 #ifdef USE_ROCM
       SmallVector<Type> retTys(nWords, IntegerType::get(getContext(), width));
-      Type retTy = retTys.size() > 1
```

Comment:
@rsanthanam-amd There is no description in the PR. Could you explain why the change works? The original logic seems to stem from NV path which uses inline asm. Since we are not using inline asm so perhaps it caused some differences and I'd like to understand why.

Reply:
For ROCm, we only emit a single load.

With the previous code, the type was specified as an aggregate if the size was greater than 1.  This was causing LLVM to give an error because it was expecting a singleton type but got an aggregate type.

This changes forces the type to always be a singleton.

Reply:
Several follow-up questions:
- Can we be sure that 1 single load would suffice?
- What would be possible values of `nWords` for different data types?
- What would be possible values of `width` for different data types?
- When `nWords` is larger than 1, the original logic creates a `LLVMStructType`, set it as `retTy`, and later on use `extract_val` to extract elements from it. New logic introduced here essentially skip those logic in "extract and store return values" section in the later part of the function.

Reply:
I can't be sure if 1 single load would suffice but since we are only doing scalar (not vectorized) loads, it would seem to be the case.

For scalar loads, nWords should always be 1.

Width is equivalent to the number of bits in the data type so it would be 8 for uint8 and 32 for float32 for example.

The dependence on LLVMStructType is not good as you suggest.  I will submit a change which removes ROCm's dependence on this.

Reply:
pushed some changes which eliminate our dependence on asmArgs